### PR TITLE
feat: add Explainer Guide on how-to-get-funding-for-open-source

### LINKED ISSUE #24: 🚀 Feature: Add a page explaining how to contribute to open source
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/OpenContributionsProject/opencontributionssite/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

Even for folks who are convinced that contributing to open source is a good idea, it can be confusing knowing how to start. And some folks want to know how contributions work before they commit to them.

Let's add a page explaining:

* Common ways of contributing money to projects
    * Direct sponsorships on GitHub, OpenCollective, etc.
    * Aggregated sponsorships on Tidelift, Thanks.dev, Stackaid, etc.
* Any common legalwork/paperwork needed to contribute employee time to open source projects
* Any common legalwork/paperwork needed to publish projects as open source

### Additional Info

Per #20, we'll eventually have a full design for the site & menus. But for now this page can live on its own.

This site is built on [Next.js](https://nextjs.org). You can create a new page by adding a `.tsx` file under `src/pages`. See https://nextjs.org/docs/basic-features/pages.

### LINKED ISSUE #111: 🚀 Feature: Create an Explainer Guide on how to get funding for open source
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/OpenContributionsProject/opencontributionssite/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

See [@JoshuaKGoldberg's comment](https://github.com/OpenContributionsProject/opencontributionssite/issues/24#issuecomment-1550322820) for reference in #24  

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by CBID2**
## PR Checklist

- [X] Addresses an existing open issue: fixes #111  
- [X] That issue was marked as [accepting prs](https://github.com/OpenContributionsProject/opencontributionssite/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [X] Steps in [CONTRIBUTING.md](https://github.com/OpenContributionsProject/opencontributionssite/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This guide teaches company leaders how they can get funding for their open source projects

### CODE REVIEW

File: src/pages/guides/How to get funding. md
Comment (JoshuaKGoldberg) [inline src/pages/guides/How to get funding. md:1]: Similar file name issue as #107 ☝️

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
I'm going to hold off reviewing this as it looks like it's two guides in one? Funding and contributing?

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,6 +4,14 @@
...
+		{
+			"rule": "*",
+			"severity": "warn"
+		}
+	],
+	"typescript.tsdk": "node_modules/typescript/lib",
+	"[jsonc]": {
+		"editor.defaultFormatter": "esbenp.prettier-vscode"
+	}
+}
```
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:17]: 🤔 err, what's going on here?

File: src/pages/guides/how-to-get-funding.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+* **GitHub Sponsors**: This is a popular option that creators use to fund their projects. It allows you to choose the number of tiers you charge and rewards you with potential sponsors. It also provides you with the opportunity to support other developers and organizations, which can be a good way to network with others and gain more support. If you’re interested in using this resource, check out the official website for more information: [https://github.com/sponsors](https://github.com/sponsors)
+* **Open Source Collective**: This is a non-profit organization that is devoted to supporting and promoting open source projects. Like GitHub sponsors, they allow you to choose the number of tiers you charge and reward you with potential sponsors. In contrast, Open Source Collective provides information on how the money is being used. If interested, check out their website for more information: [https://opencollective.com/how-it-works](https://opencollective.com/how-it-works)
```
Comment (JoshuaKGoldberg) [inline src/pages/guides/how-to-get-funding.md]: [Proofreading] A few issues here:

* There's no "Source" in their name..
* They're a company, not a non-profit.
* I find the phrasing _"reward(s) you with"_ to be a bit confusing. Is it that you may potentially get the reward of sponsors joining? I'd think something like _"allow"_ or _"enable"_ might be a better word here than _"reward"_.
* I think this should also mention [StackAid](https://www.stackaid.us), [Thanks.dev](https://thanks.dev), and [Tidelift](https://tidelift.com/lifter)

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
I started reviewing and don't have time for a full review - hopefully these comments so far are helpful!

#107 shares some styles of build failures with this one (e.g. _Lint Markdown_). Please fix those and I can re-review. Thanks!

File: src/pages/guides/how-to-get-funding.md (hunk 2 above)
Comment (CBID2) [reply src/pages/guides/how-to-get-funding.md]: Thanks for the feedback @JoshuaKGoldberg! I did mention the sponsors you listed. I just added them in the aggregated section.

**Comment by CBID2** [comment]
@JoshuaKGoldberg, for some reason, [the Lint Spelling test](https://github.com/OpenContributionsProject/opencontributionssite/actions/runs/5006864777/jobs/8974370495?pr=109) keeps saying that Stackaid is an unknown error. How can I solve this?

**Comment by JoshuaKGoldberg** [comment]
Great question - per the [CSpell docs](https://cspell.org) > [`cspell.json`](https://cspell.org/configuration/#cspelljson), it can be added to the `words` array in that file.

File: .vscode/settings.json [hunk 3]
```diff
@@ -4,6 +4,14 @@
...
+	"eslint.rules.customizations": [
+		{
+			"rule": "*",
+			"severity": "warn"
+		}
+	],
+	"typescript.tsdk": "node_modules/typescript/lib",
+	"[jsonc]": {
+		"editor.defaultFormatter": "esbenp.prettier-vscode"
+	}
```
Comment (JoshuaKGoldberg) [inline .vscode/settings.json:16]: [Request] Can you revert these unrelated changes please? If you want to add them in we can discuss in a new issue.

File: src/pages/guides/how-to-get-funding-for-your-open-source-projects.md. md
Comment (JoshuaKGoldberg) [inline src/pages/guides/how-to-get-funding-for-your-open-source-projects.md. md:1]: Looks like there's an extra ` md` (notice the space) in this file name. You'll want to remove that and then fix up the formatting & linting errors.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks again for sending & for the ping! Just a bit of stuff to get through first. ✨

File: src/pages/guides/how-to-get-funding-for-your-open-source-projects.md. md [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# How To Get Funding For Your Open Source Projects
+
+
+# Overview:
```
Comment (blackgirlbytes) [inline src/pages/guides/how-to-get-funding-for-your-open-source-projects.md. md:4]: I'm not a super huge fan of having a h1 after another h1 for accessibility screenreader reasons, read [here](https://blog.shimin.io/should-you-use-multiple-h1-s-in-2022/#:~:text=Using%20more%20than%20one%20%3Ch1,is%20beneficial%20for%20screenreader%20users.)

**Review by blackgirlbytes — CHANGES_REQUESTED** [review]

**Comment by JoshuaKGoldberg** [comment]
Talking privately - closing this in favor of #141
